src/core/net/dns_headers.hpp

```diff
@@ -735,6 +735,17 @@ class Name
      */
     static otError CompareName(const Message &aMessage, uint16_t &aOffset, const Message &aMessage2, uint16_t aOffset2);
 
+    /**
+     * This static method tests if a DNS name ends with given domain.
+     *
+     * @param[in]  aName    The dot-separated name.
```

Comment:
I think both `aName` and `aDomain` should end with dot. 
Make it clear in doxygen?

Reply:
done.